activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -220,9 +220,11 @@ def explain(arel, binds = [])
 
       # Executes the SQL statement in the context of this connection.
       def execute(sql, name = nil)
-        log(sql, name) do
-          ActiveSupport::Dependencies.interlock.permit_concurrent_loads do
-            @connection.query(sql)
+        with_connection_verify(::ActiveRecord::StatementInvalid) do
+          log(sql, name) do
```

Comment:
Is the intention to log every attempt?  Since this does the error translation that is causing you to pass a different exception class between this method and `quote_string`.